pkg/pools/pools.go

```diff
@@ -126,12 +135,23 @@ func (bufPool *BufioWriterPool) Put(b *bufio.Writer) {
 // NewWriteCloserWrapper returns a wrapper which puts the bufio.Writer back
 // into the pool and closes the writer if it's an io.Writecloser.
 func (bufPool *BufioWriterPool) NewWriteCloserWrapper(buf *bufio.Writer, w io.Writer) io.WriteCloser {
+	closed := false
 	return ioutils.NewWriteCloserWrapper(w, func() error {
+		if closed {
+			subsequentCloseWarn("WriteCloserWrapper")
+			return nil
+		}
 		buf.Flush()
 		if writeCloser, ok := w.(io.WriteCloser); ok {
 			writeCloser.Close()
 		}
 		bufPool.Put(buf)
+		closed = true
 		return nil
 	})
 }
+
+func subsequentCloseWarn(name string) {
+	log.G(context.TODO()).Warn("subsequent attempt to close " + name)
+	log.G(context.TODO()).Debugf("stack trace: %s", string(debug.Stack()))
```

Comment:
Is the `string(debug.Stack()` always evaluated here? (regardless if "debug" logging is enabled)?

If so, we should probably gate this line based on the current debug-level (or would including it in `WithFields()` do the trick?)

Reply:
Unfortunately yes, `WithFields` won't solve this as we need to explicitly convert the `[]byte` to a string, otherwise it will be logged in a binary form (array of bytes like `[255, 0, 1, 5, ...]`).

I can do explicit level check.

Reply:
Yeah, maybe we need that; I must say that I didn't look deep into how "costly" this part is, but  I _think_ this code could be in a "hot" path, so it may be relevant if it is.